docs/src/userguide/config/event_config.rst

```diff
@@ -126,13 +126,176 @@ Example configuration for a custom event input that is not supported by Klio:
+          streaming: True
+        job_config:
+        events:
+            inputs:
+            - type: bq
+              query:
+                SELECT * FROM [gcp_project.bigquery_dataset.bigquery_table]
+
+.. option:: job_config.events.inputs[].type
+
+    Value: ``bq``
+
+    | **Runner**: Dataflow, Direct
+    | *Required*
+
+.. option:: job_config.events.inputs[].columns[]
+
+    Not applicable if query is specified.
+
+    A list of strings specifying the columns to read events from.
+    If only one column is provided, the value will be returned as a bytestring.
+    If no columns are specified, or if more than one column is specified,
+    the results including the column names will be serialized to a JSON bytestring,
+    e.g. ``'{"field1": "foo", "field2": bar"}'``.
+
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].columns[].<column> STR
+
+    A column in the table or query result used to build the event input from.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].table STR
+
+    Name of the event input table.
+
+    If specified all data of the table will be used as input of the current source.
+    The ID must contain only letters ``a-z``, ``A-Z``, numbers ``0-9``,
+    or underscores ``_``.
+
+    If dataset and query arguments are not specified,  then the table argument must
+    contain the entire table reference specified as:
+    ``'DATASET.TABLE'`` or ``'PROJECT:DATASET.TABLE'``.
+
+    | **Runner**: Dataflow, Direct
+    | *Required unless query provided*
+
+.. option:: job_config.events.inputs[].dataset STR
+
+    Name of the event input table's dataset.
+
+    Ignored if the table reference is fully specified by the table argument or
+    if a query is specified.
+
+    | **Runner**: Dataflow, Direct
+    | *Required if query not provided and table reference does not include dataset*
+
+.. option:: job_config.events.inputs[].project STR
+
+    Name of the event input table's project.
+
+    Ignored if the table reference is fully specified by the table argument or
+    if a query is specified.
+
+    | **Runner**: Dataflow, Direct
+    | *Required if query is not provided and table reference does not include project*
+
+.. option:: job_config.events.inputs[].query STR
+
+    Query string supplying the columns. Mutually exclusive with specifying the
+    table field.
+
+    | **Runner**: Dataflow, Direct
+    | *Required if table is not specified*
+
+.. option:: job_config.events.inputs[].validate BOOL
+
+    If :data:`True`, various checks will be done when source
+    gets initialized (e.g., is table present?). This should be
+    :data:`True` for most scenarios in order to catch errors as early as
+    possible (pipeline construction instead of pipeline execution). It
+    should be :data:`False` if the table is created during pipeline
+    execution by a previous step. Defaults to :data:`False`.
```

Comment:
This actually defaults to True ([see here](https://github.com/spotify/klio/blob/master/core/src/klio_core/config/_io.py#L381)).